.github/workflows/build-push-pr.yml

```diff
@@ -1,6 +1,6 @@
 name: Build container image for PR
 on:
-  pull_request:
+  pull_request_target:
```

Comment:
I am not completely sure about the security implications of that. Apart from this, the PR looks good to me.

Reply:
This is described here: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target

The warning explains that this is run in the context of the target branch of the PR (in our case, the `main` branch), meaning it will get access to the organisation secrets needed to use the external docker builder.

But I think this is safe, as this workflow is only triggered if there is a specific label on the PR, which can only be set if you have permissions on the repo.

Reply:
Right, but someone could always push to the PR after the label is granted. It may be safer to only run it from in-repo branches.

Reply:
Ok, I updated the PR so the PR branch needs to be from the target repository (`mastodon/mastodon` in our case).